src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Generator.cs

```diff
@@ -185,15 +195,13 @@ public GenerationResult GenerateJsonDocument(SbomPackage packageInfo)
 
     private bool ShouldWeKeepTheExistingId(string spdxId)
     {
-        switch (configuration.ManifestToolAction)
```

Comment:
Quick note: Prior to #1139, this code always checked for equality with `Constants.RootPackageIdValue`. In that sense, the code in this PR gets closer to the code before #1139.